src/System Application/App/Json/src/JsonImpl.Codeunit.al

```diff
@@ -0,0 +1,417 @@
+    var
+        RecID: RecordID;
+        Value: Variant;
+        IntVar: Integer;
+        DecimalVal: Decimal;
+        GuidVal: Guid;
+        DateVal: Date;
+        BoolVal, Success : Boolean;
+        JPropertyDotNet: DotNet JProperty;
+    begin
+        Success := false;
+        JPropertyDotNet := JsonObjectDotNet.SelectToken(propertyPath);
+
+        if IsNull(JPropertyDotNet) then
+            exit(false);
+
+        Value := Format(JPropertyDotNet.Value, 0, 9);
+
+        case FieldRef.Type of
+            FieldType::Integer,
+            FieldType::Decimal:
+                begin
+                    Success := Evaluate(DecimalVal, Value, 9);
+                    FieldRef.Value(DecimalVal);
+                end;
+            FieldType::Date:
+                begin
+                    Success := Evaluate(DateVal, Value, 9);
+                    FieldRef.Value(DateVal);
+                end;
+            FieldType::Boolean:
+                begin
+                    Success := Evaluate(BoolVal, Value, 9);
+                    FieldRef.Value(BoolVal);
+                end;
+            FieldType::GUID:
+                begin
+                    Success := Evaluate(GuidVal, Value);
+                    FieldRef.Value(GuidVal);
+                end;
+            FieldType::Text,
+            FieldType::Code:
+                begin
+                    FieldRef.Value(CopyStr(Value, 1, FieldRef.Length));
+                    Success := true;
+                end;
+            FieldType::Option:
+                begin
+                    if not Evaluate(IntVar, Value) then
+                        IntVar := TextToOptionValue(Value, FieldRef.OptionCaption);
+                    if IntVar >= 0 then begin
+                        FieldRef.Value := IntVar;
+                        Success := true;
+                    end;
+                end;
+            FieldType::BLOB:
+                if TryReadAsBase64(FieldRef, Value) then
+                    Success := true;
+            FieldType::RecordID:
+                begin
+                    Success := Evaluate(RecID, Value);
+                    FieldRef.Value(RecID);
+                end;
+        end;
+
+        exit(Success);
+    end;
+
+    [TryFunction]
+    local procedure TryReadAsBase64(var BlobFieldRef: FieldRef; Value: Text)
+    var
+        Base64Convert: Codeunit "Base64 Convert";
+        TempBlob: Codeunit "Temp Blob";
+        RecordRef: RecordRef;
+        OutStream: OutStream;
+    begin
+        TempBlob.CreateOutStream(OutStream);
+        Base64Convert.FromBase64(Value, OutStream);
+        RecordRef := BlobFieldRef.Record();
+        TempBlob.ToRecordRef(RecordRef, BlobFieldRef.Number);
+    end;
+
+    local procedure TextToOptionValue(InputText: Text; OptionString: Text): Integer
+    var
+        IntVar: Integer;
+        Counter: Integer;
+    begin
+        if InputText = '' then
+            InputText := ' ';
```

Comment:
Is this for making the `evaluate` to fail?

Reply:
Yep, so it go to the `end else begin` path. This function is taken from the `codeunit 5302 "Outlook Synch. Type Conv" TextToOptionValue()`